catroid/src/main/java/org/catrobat/catroid/ui/SpriteActivity.java

```diff
@@ -235,6 +235,12 @@ public void onBackPressed() {
 			getSupportFragmentManager().popBackStack();
 			return;
 		}
+		if (getCurrentFragment() instanceof ScriptFragment
```

Comment:
I'm pretty sure this should be before `if (getSupportFragmentManager().getBackStackEntryCount() > 0) { ..` because if there is something on the back stack pressing the back button will not cancel the highlighting. you could probably combine it with the `isCurrentlyMoving(..)` call above, because for both states you will have to be in the `ScriptFragment`.

I'd suggest something like this:
```
Fragment currentFragment = getCurrentFragment();

if (currentFragment instanceof ScriptFragment) {
    if (((DragAndDropInterface) getCurrentFragment()).isCurrentlyMoving()) {
        ((DragAndDropInterface) getCurrentFragment()).cancelMove();
        return;
    }
    if (((ScriptFragment) getCurrentFragment()).isCurrentlyHighlighted())) {
        ((ScriptFragment) getCurrentFragment()).cancelHighlighting();
        return;
    }
```

although I'm not 100 % sure if that is the correct behaviour now. can highlighting happen at the same time as drag and drop?

Reply:
highlighting and drag and drop at the same time should not be possible since tapping on something (when highlighting is active) stops the highlighting of the brick/bricks and executes the tapped.

Reply:
ok, nice.